pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -255,7 +255,7 @@ func buildLogsQuery(panelType v3.PanelType, start, end, step int64, mq *v3.Build
 			"SELECT now() as ts,"
 		// step or aggregate interval is whole time period in case of table panel
 		step = (utils.GetEpochNanoSecs(end) - utils.GetEpochNanoSecs(start)) / 1000000000
-	} else if panelType == v3.PanelTypeGraph || panelType == v3.PanelTypeValue {
+	} else if panelType == v3.PanelTypeGraph || panelType == v3.PanelTypeBar || panelType == v3.PanelTypeValue {
```

Comment:
Can this be `else` instead of `else if` that works for the rest of the panel types too?

Reply:
Don't want to change this as it might have unintended changes. For now let's keep it as it is.

Reply:
Please help me understand unintended things could happen. What worked will continue to work because nothing is being removed and every other panel type is an aggregation type so I want to understand what will be the issue.

Reply:
I would suggest taking time to fix this so that we don't get invalid query alerts. Meanwhile, I will fix the panel type in query builder alerts because rn we only support time series and there is no issue in fixing it in alerts.

Reply:
> Please help me understand unintended things could happen. What worked will continue to work because nothing is being removed and every other panel type is an aggregation type so I want to understand what will be the issue.

I was worried about list panel type case, I just checked and verified we are overwriting that so this case can be made.

Reply:
> I would suggest taking time to fix this so that we don't get invalid query alerts. Meanwhile, I will fix the panel type in query builder alerts because rn we only support time series and there is no issue in fixing it in alerts.

We have already addressed issue on frontend now. So creation of new alerts via UI should only be possible for time series. We can make changes on backend as well to prevent via API.